pass-nihms-loader/nihms-data-transform-load/src/test/java/org/eclipse/pass/loader/nihms/entrez/EntrezPmidLookupTest.java

```diff
@@ -15,41 +15,96 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.TestPropertySource;
 
 /**
  * @author Karen Hanson
- * @version $Id$
  */
+@SpringBootTest
+@TestPropertySource("classpath:test-application.properties")
+@WireMockTest(httpPort = 9911)
 public class EntrezPmidLookupTest {
 
+    @Autowired
+    protected PmidLookup pmidLookup;
+
+    // Needed so tests can run after application starts
+    @MockBean private NihmsTransformLoadCLIRunner nihmsTransformLoadCLIRunner;
+
+    @Value("${pmc.entrez.service.url}")
+    private String ENTREZ_PATH;
+
     @Test
-    public void testGetEntrezRecordJson() {
-        PmidLookup apiService = new PmidLookup();
+    public void testGetEntrezRecordJson() throws IOException, URISyntaxException {
+        String entrez = IOUtils.toString(getClass().getClassLoader().
+                getResourceAsStream("pmidrecord.json"), StandardCharsets.UTF_8);
 
-        String pmid = "29249144";
+        String pmid = "11111111";
 
-        JSONObject pmr = apiService.retrievePubMedRecordAsJson(pmid);
-        assertTrue(pmr.getString("source").contains("Proteome"));
+        stubFor(get(urlPathEqualTo("/entrez/eutils/esummary.fcgi"))
+                .withQueryParam("db", WireMock.equalTo("pubmed"))
+                .withQueryParam("retmode", WireMock.equalTo("json"))
+                .withQueryParam("rettype", WireMock.equalTo("abstract"))
+                .withQueryParam("id", WireMock.equalTo(pmid))
+                .willReturn(aResponse().withStatus(200).withBody(entrez)));
 
+        JSONObject pmr = pmidLookup.retrievePubMedRecordAsJson(pmid);
+        assertTrue(pmr.getString("source").contains("Journal A"));
     }
 
     @Test
-    public void testGetPubMedRecord() {
-        PmidLookup pmidLookup = new PmidLookup();
-        String pmid = "29249144";
+    public void testGetPubMedRecord() throws IOException {
+        String entrez = IOUtils.toString(getClass().getClassLoader().
+                getResourceAsStream("pmidrecord.json"), StandardCharsets.UTF_8);
+
+        String pmid = "11111111";
+
+        stubFor(get(urlPathEqualTo("/entrez/eutils/esummary.fcgi"))
+                .withQueryParam("db", WireMock.equalTo("pubmed"))
+                .withQueryParam("retmode", WireMock.equalTo("json"))
+                .withQueryParam("rettype", WireMock.equalTo("abstract"))
+                .withQueryParam("id", WireMock.equalTo(pmid))
+                .willReturn(aResponse().withStatus(200).withBody(entrez)));
+
         PubMedEntrezRecord record = pmidLookup.retrievePubMedRecord(pmid);
-        assertEquals("10.1021/acs.jproteome.7b00775", record.getDoi());
+        assertEquals("10.1000/a.abcd.1234", record.getDoi());
     }
 
     @Test
-    public void testGetPubMedRecordWithHighAsciiChars() {
-        PmidLookup pmidLookup = new PmidLookup();
-        String pmid = "27648456";
+    public void testGetPubMedRecordWithHighAsciiChars() throws IOException {
```

Comment:
A suggestion to reduce code is to update `pmidrecord.json` to have the high ascii character in the  `pmid_record_ascii.json` `title`.  Then you can just verify the title in `testGetPubMedRecord` and delete `testGetPubMedRecordWithHighAsciiChars`.

Reply:
Other tests use this pmidrecord.json such as `testNewCompliantPublication`, `testNewInProcessPublication`, `testNewNonCompliantPublication`. Would it be okay to change these tests as well with the high ascii character?

Reply:
These are in `TransformAndLoadCompliantIT`, `TransformAndLoadInProcessIT`, and `TransformAndLoadNonCompliantIT`

Reply:
Done.